docs/spatial-index-guide.md

```diff
@@ -0,0 +1,94 @@
+<!--
+    Licensed to the Apache Software Foundation (ASF) under one or more 
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership. 
+    The ASF licenses this file to you under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with 
+    the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+    
+    Unless required by applicable law or agreed to in writing, software 
+    distributed under the License is distributed on an "AS IS" BASIS, 
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and 
+    limitations under the License.
+-->
+
+## What is spatial index
+
+A spatial index is a data structure that allows for accessing a spatial object efficiently. It is a common technique used by spatial databases.  Without indexing, any search for a feature would require a "sequential scan" of every record in the database, resulting in much longer processing time. In a spatial index construction process, the minimum bounding rectangle serves as an object approximation. Various types of spatial indices across commercial and open-source databases yield measurable performance differences. Spatial indexing techniques are playing a central role in time-critical applications and the manipulation of spatial big data.
+
+
+
+## What does carbondata implement spatial index
+
+There are many components that implement spatial indexing, like GeoSpark that use GeoMesa format for spatial query. now carbondata implements  a different way of spatial index, more like an UDF.  Its core is to use grid coordinates to generate coordinate based hash ID, like Z order, it's also regionally continuous.
```

Comment:
Have read the text and have few suggestions on repharsing certain parts of it, To be clear, have attached the modified text below. Can we rephrase like this -->
There are many opensource implementations for spatial indexing and to process spatial queries. CarbonData implements a different way of spatial index. Its core idea is to use the raster data. Raster is made up of matrix of cells organized into rows and columns(called a grid). Each cell represents a coordinate. And the index for the coodrinate is generated using longitude and latitude pair, like the Z order curve.

                  CarbonData implements a grid spatial index. It requires rasterization of data before loading into segments. A set of latitude and longitude represents a grid range. The size of the grid can be configured. So the coordinates of the loaded points are often discrete and not continuous.

Reply:
thank you for your help.
has replace the words.